ext/afform/core/Civi/Api4/Action/Afform/Submit.php

```diff
@@ -483,12 +483,23 @@ protected static function saveJoins(AfformSubmitEvent $event, $index, $entityId,
       // the contact was being auto-updated via a dedupe rule; in that case we would not want to
       // delete any existing records.
       elseif ($values) {
-        $result = civicrm_api4($joinEntityName, 'replace', [
-          // Disable permission checks because the main entity has already been vetted
-          'checkPermissions' => FALSE,
-          'where' => $whereClause,
-          'records' => $values,
-        ]);
+        // Based on afform Submit::getCustomGroupBlocks(), we can use this check for "is_multiple"
+        if (str_contains($joinEntityName, "Custom_")) {
+          $values[0]["entity_id"] = $entityId;
+          $result = civicrm_api4($joinEntityName, 'save', [
```

Comment:
We want to propagate the `$entityId` into all records, not just the first one. APIv4 has a shorthand way of doing that:
```suggestion
          $result = civicrm_api4($joinEntityName, 'save', [
            'defaults' = ['entity_id' => $entityId],
```